### PR TITLE
Implemented Assignment questions 17 and 19

### PR DESCRIPTION
**Pull request by shravan-cc**
This PR introduces two modules: alphabetProcessor and stringProcessor, providing essential functions for alphabet and string processing, respectively. The alphabetProcessor module offers methods to generate lowercase alphabets and categorize them into vowels and consonants. On the other hand, the stringProcessor module includes functions to trim leading and trailing whitespace from strings, as well as reduce consecutive whitespace characters to a single space. Additionally, both modules feature a compose function for combining multiple processing steps into a single function. The accompanying test case ensures the correctness of the string processing functionalities.

### CODE REVIEW

File: assignment.js [hunk 1]
```diff
@@ -597,3 +597,89 @@ export function numberTransformer() {
     compose,
   };
 }
+/**
+ * This module provides functions to generate and process alphabets.
+ */
+export function alphabetProcessor() {
```
Comment (shripada) [inline assignment.js]: Avoid such wrappers, unless you want to close over some private data. This introduces an unwanted layer, which does nothing much.

File: assignment.js [hunk 2]
```diff
@@ -597,3 +597,89 @@ export function numberTransformer() {
...
+   */
+  const compose =
+    (...functions) =>
+    (arguements) =>
+      functions.reduceRight((acc, func) => func(acc), arguements);
+  return {
+    trimLeading,
+    trimTrailing,
+    singleSpace,
+    compose,
```
Comment (shripada) [inline assignment.js]: look we already duplicating compose, because of unwanted layers

File: assignment.js [hunk 3]
```diff
@@ -597,3 +597,89 @@ export function numberTransformer() {
...
+    generateAlphabets,
+    categorizeAlphabets,
+    compose,
+  };
+}
+/**
+ * A module for string processing utility functions.
+ * @module stringProcessor
+ */
+export function stringProcessor() {
```
Comment (shripada) [inline assignment.js]: same feedback

**Review by shripada — CHANGES_REQUESTED** [review]

File: assignment.js (hunk 1 above)
Comment (shravan-cc) [reply assignment.js]: Okay Sir.Addressed the changes.

File: assignment.js (hunk 3 above)
Comment (shravan-cc) [reply assignment.js]: Implemented the changes

File: assignment.js (hunk 2 above)
Comment (shravan-cc) [reply assignment.js]: Implemented the changes

**Review by shripada — APPROVED** [review]

**Pull request merged**